napari/_tests/test_conftest_fixtures.py

```diff
@@ -65,7 +65,7 @@ def f() -> str:
 
 @patch('qtpy.QtCore.QTimer.start')
 @patch('qtpy.QtCore.QTimer.isActive', return_value=True)
-def test_lack_disable_throttle(_active_mock, start_mock, monkeypatch):
```

Comment:
The failure comes from removing `_active_mock` from the signature. It is required to start_mock be a proper patch object.

Reply:
It's weird... the linter complains if I readd it:

```py
napari/_tests/test_conftest_fixtures.py:68:32: PT019 Fixture `_active_mock` without value is injected as parameter, use `@pytest.mark.usefixtures` instead
```

so I add it using `@pytest.mark.usefixtures`, but then I get this when running tests:

```py
file /home/lorenzo/git/napari/napari/_tests/test_conftest_fixtures.py, line 66
  @pytest.mark.usefixtures('_active_mock')
  @patch('qtpy.QtCore.QTimer.start')
  @patch('qtpy.QtCore.QTimer.isActive', return_value=True)
  def test_lack_disable_throttle(start_mock, monkeypatch):
E       fixture '_active_mock' not found
>       available fixtures: MouseEvent, _auto_shutdown_dask_threadworkers, _clean_themes, _dangling_qanimations, _dangling_qthread_pool, _dangling_qthreads, _dangling_qtimers, _disable_notification_dismiss_timer, _disable_throttling, _fresh_settings, _no_error_reports, _skip_examples, add_implementation, add_specification, builtins, cache, caller_from_implementation, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, layer, layer_data_and_types, layers, make_napari_viewer, make_napari_viewer_proxy, mock_app, monkeypatch, napari_plugin_manager, napari_svg_name, npe2pm, npe2pm_, pytestconfig, pytester_pretty, qapp, qapp_args, qapp_cls, qtbot, qtlog, qtmodeltester, record_property, record_testsuite_property, record_xml_attribute, recwarn, single_threaded_executor, temporary_hookimpl, test_plugin_manager, tmp_path, tmp_path_factory, tmp_plugin, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.
```

And that's why I removed it, cause I thought it was a leftover mistake.

Reply:
`_active_mock` and `start_mock` are not fixtures, but output of `@patch('qtpy.QtCore.QTimer.start')` and `@patch('qtpy.QtCore.QTimer.isActive', return_value=True)`. 

To fix this, I think that patching need to be done using monkeypatch.

Reply:
oh... Ok, I don't know how `@patch` works ^^' So the linter just thinks it's a fixture but it's not. Maybe I should just `noqa` it then?

Reply:
I have pushed commit with test refactor